x-pack/plugins/task_manager/server/task_runner.ts

```diff
@@ -261,12 +278,12 @@ export class TaskManagerRunner implements TaskRunner {
    */
   public async cancel() {
     const { task } = this;
-    if (task && task.cancel) {
+    if (task?.cancel) {
       this.task = undefined;
       return task.cancel();
     }
 
-    this.logger.warn(`The task ${this} is not cancellable.`);
+    this.logger.debug(`The task ${this} is not cancellable.`);
```

Comment:
After seeing this in the console a few times, from telemetry tasks from various apps, seems correct to change this to a debug rather than warn, since it's noise at this point.

**Howevers** ... I think we should actually enforce that tasks provide a `cancel()` method.  Runaway tasks will occur, and we should provide the ability for clients to clean themselves up, and so by forcing them to implement it, they'll at least need to think about it.  Out of scope for this issue, and I believe we have a very old issue open concerning cancellation anyway ... hmmm, can't find one - I think we should open one, if no one else can find one either.  We should support cancellation for both alerts and actions as run by TM, and enhance our current actions and alerts to do the right thing.

Reply:
That's a fair point and worth investigating - I'll add an issue